Source/WebKit/Platform/IPC/unix/ConnectionUnix.cpp

```diff
@@ -554,23 +554,27 @@ SocketPair createPlatformConnection(unsigned options)
 
         if (!(options & SetCloexecOnServer))
             RELEASE_ASSERT(unsetCloseOnExec(sockets[1]));
-
         if (!(options & SetCloexecOnClient))
             RELEASE_ASSERT(unsetCloseOnExec(sockets[0]));
 
-        SocketPair socketPair = { sockets[0], sockets[1] };
-        return socketPair;
+        goto out;
```

Comment:
Can we do the setsockopt in a helper function/lambda instead of using goto?

Reply:
Probably, I'll try